internal/policy/root_test.go

```diff
@@ -29,6 +29,66 @@ func TestInitializeRootMetadata(t *testing.T) {
 	assert.Equal(t, []string{key.KeyID}, rootMetadata.Roles[RootRoleName].KeyIDs)
 }
 
+func TestAddRootKey(t *testing.T) {
+	keyBytes, err := os.ReadFile(filepath.Join("test-data", "root.pub"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	key, err := tuf.LoadKeyFromBytes(keyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rootMetadata := InitializeRootMetadata(key)
+
+	newRootKeyBytes, err := os.ReadFile(filepath.Join("test-data", "targets-1.pub"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	newRootKey, err := tuf.LoadKeyFromBytes(newRootKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rootMetadata = AddRootKey(rootMetadata, newRootKey)
+
+	assert.Equal(t, newRootKey, rootMetadata.Keys[newRootKey.KeyID])
+	assert.Equal(t, []string{key.KeyID, newRootKey.KeyID}, rootMetadata.Roles[RootRoleName].KeyIDs)
+}
+
+func TestRemoveRootKey(t *testing.T) {
+	keyBytes, err := os.ReadFile(filepath.Join("test-data", "root.pub"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	key, err := tuf.LoadKeyFromBytes(keyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rootMetadata := InitializeRootMetadata(key)
+
+	newRootKeyBytes, err := os.ReadFile(filepath.Join("test-data", "targets-1.pub"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	newRootKey, err := tuf.LoadKeyFromBytes(newRootKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rootMetadata = AddRootKey(rootMetadata, newRootKey)
+
+	rootMetadata, err = RemoveRootKey(rootMetadata, newRootKey.KeyID)
+
+	assert.Nil(t, err)
+	assert.Equal(t, key, rootMetadata.Keys[key.KeyID])
+	assert.Equal(t, newRootKey, rootMetadata.Keys[newRootKey.KeyID])
+	assert.Equal(t, []string{key.KeyID}, rootMetadata.Roles[RootRoleName].KeyIDs)
```

Comment:
Can you try removing the original key too and trigger ErrCannotMeetThreshold?

Reply:
Added